packages/iov-bcp-types/src/transactions.ts

```diff
@@ -44,131 +37,62 @@ export interface Amount {
   readonly tokenTicker: TokenTicker;
 }
 
-export interface ChainAddressPair {
-  readonly chainId: ChainId;
-  readonly address: Address;
-}
-
-export enum TransactionKind {
-  AddAddressToUsername,
-  Send,
-  /** @deprecated see SetNameTx */
-  SetName,
-  SwapOffer,
-  SwapCounter,
-  SwapClaim,
-  SwapTimeout,
-  RegisterBlockchain,
-  RegisterUsername,
-  RemoveAddressFromUsername,
-}
-
-export interface BaseTx {
+/** The basic transaction type all transactions should extend */
+export interface UnsignedTransaction {
+  /**
+   * The domain in which the concrete transaction is valid
+   *
+   * This should be used for type detection only and not be encoded somewhere.
+   * Right now we use "bns", "ethereum", "lisk" and "rise" but this could also
+   * be migrated to a Java-style package names like "io.lisk.mainnet" later on.
+   */
+  readonly domain: string;
```

Comment:
Okay.... So I need to change this to use SendTransaction on lisk vs ethereum.

If we could add a `bcp` domain for all transactions defined in `iov-bcp-types` it seems it would be cleaner to allow the exact same transaction be cross-chain, while allowing each chain to carve out it's own extensions.

I need to see examples